lib/hammer_cli/defaults.rb

```diff
@@ -0,0 +1,194 @@
+require 'hammer_cli'
+require 'yaml'
+module HammerCLI
+
+  class BaseDefaultsProvider
```

Comment:
Was there any reason for keeping all methods at class level and avoid creating instances? Maybe I overlooked something but it seems to me that apart from `register_provider` they all can be instance methods.